gcs/src/components/config/motorTest.jsx

```diff
@@ -13,14 +15,69 @@ import tailwindConfig from '../../../tailwind.config'
 const tailwindColors = resolveConfig(tailwindConfig).theme.colors
 
 // Custom helper function
+import { useLocalStorage } from '@mantine/hooks'
+import { FRAME_CLASS_MAP, LETTERS } from '../../helpers/mavlinkConstants'
 import { socket } from '../../helpers/socket'
 
-export default function Motortestpanel({
-  selectedThrottle,
-  selectedDuration,
-  setSelectedThrottle,
-  setSelectedDuration,
-}) {
+export default function Motortestpanel() {
+  const [connected] = useLocalStorage({
+    key: 'connectedToDrone',
+    defaultValue: false,
+  })
+  const [frameTypeOrder, setFrameTypeOrder] = useState(null)
+  const [frameTypeDirection, setFrameTypeDirection] = useState(null)
+  const [frameTypename, setFrameTypename] = useState(null)
+  const [frameClass, setFrameClass] = useState(null)
+  const [numberOfMotors, setNumberOfMotors] = useState(4)
+  const [selectedThrottle, setSelectedThrottle] = useState(10)
+  const [selectedDuration, setSelectedDuration] = useState(2)
+
+  useEffect(() => {
+    if (!connected) {
+      return
+    } else {
+      socket.emit('set_state', { state: 'config.motor_test' })
+      socket.emit('get_frame_config')
+    }
+    socket.on('frame_type_config', (data) => {
+      let currentFrameType = data.frame_type
+      let currentFrameClass = data.frame_class
+      console.log('got here')
+
+      // Checks if the frame class has any compatible frame types and if the current frame type param is comaptible
+      if (FRAME_CLASS_MAP[currentFrameClass].frametype) {
+        if (
+          Object.keys(FRAME_CLASS_MAP[currentFrameClass].frametype).includes(
+            currentFrameType.toString(),
+          )
+        ) {
+          setFrameTypeDirection(
```

Comment:
Here maybe create a new variable which stores the frame information, e.g. `const frameInfo = FRAME_CLASS_MAP[currentFrameClass].frametype[currentFrameType]`. Then you can easily access this below.